### PR TITLE
Added areasearch application page and components and routes

### PR DESCRIPTION
**Pull request by PasiVuohijoki**
Backend branch feature-areasearch-form should be used.

### CODE REVIEW

**Comment by EmiliaMakelaVincit** [comment]
The share of rent field type still shows as unimplemented on the application view.

File: src/areaSearch/areaSearchSpecsPage.tsx [hunk 1]
```diff
@@ -459,5 +473,6 @@ export default connect(
     setSubmitSucceeded,
     touch,
     fetchIntendedUses,
+    changeForm: change,
```
[the remark this answers is not in the document]
Comment (PasiVuohijoki) [reply src/areaSearch/areaSearchSpecsPage.tsx:476]: There are "change" -function already at the InjectedFormProps and it has different parameters. This is why I aliased it.

File: src/application/components/applicationPreviewSubsectionField.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      default:
+        // treat unknown types as plain text
+        break;
+    }
+
+    return displayValue;
+  };
+
+  return (
+    <dl className="ApplicationPreviewPage__subsection-fields">
```
[the remark this answers is not in the document]
Comment (EmiliaMakelaVincit) [reply src/application/components/applicationPreviewSubsectionField.tsx]: This is still a bit off. The name of the component is `ApplicationPreviewSubsectionField`, thus the class of the main element should be `ApplicationPreviewSubsectionField` as well.

File: src/application/components/applicationPreviewSubsection.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    if (parentApplicantType) {
+      return parentApplicantType;
+    }
+
+    return ApplicantTypes.BOTH;
+  };
+
+  return (
+    <div className="ApplicationPreviewPage__subsection">
```
[the remark this answers is not in the document]
Comment (EmiliaMakelaVincit) [reply src/application/components/applicationPreviewSubsection.tsx]: And `ApplicationPreviewSubsection` here.

File: src/application/components/_applicationFractionalField.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
+.ApplicationFractionField {
+  fieldset {
+    display: flex;
+    margin-inline-end: 0;
+    margin-inline-start: 0;
```
Comment (EmiliaMakelaVincit) [inline src/application/components/_applicationFractionalField.scss:5]: TIL that these exist. Not exactly a bleeding edge feature either (added to FF and Chrome in '19-'20), so it should probably be fine to use.

File: src/areaSearch/areaSearchSpecsPage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (EmiliaMakelaVincit) [reply src/areaSearch/areaSearchSpecsPage.tsx:488]: Hmm, I see. Using InjectedFormProps in the first place looks like my mistake as I think the component doesn't actually ever get most of the props included in it. (It doesn't get `redux-form`'d directly, and it only receives `valid` as a manually handed down prop from above.)

**Review by EmiliaMakelaVincit — COMMENTED** [review]
I think I've reviewed everything except the attachments part now; I'll do those separately next.

File: src/application/components/applicationFractionalFieldset.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+          className="ApplicationFractionField__field"
+          id={`${id}-denominator`}
+          value={parseValue(FieldType.DENOMINATOR)}
+          onChange={(e) => changeHandler(e.target.value, FieldType.DENOMINATOR)}
+          invalid={meta.invalid}
+          required={field.required}
+          errorText={meta.error}
+          helperText={field.hint_text}
+        />
+      </Fieldset>
```
Comment (EmiliaMakelaVincit) [inline src/application/components/applicationFractionalFieldset.tsx:62]: Optimally, these fields should take numbers only. (Adding `type="number"` might be enough if HDS is smart about it.)

**Review by EmiliaMakelaVincit — COMMENTED** [review]
...One more comment that I accidentally deleted before sending it in.

File: src/areaSearch/areaSearchSpecsPage.tsx [hunk 6]
```diff
@@ -156,13 +178,15 @@ const AreaSearchSpecsPage = ({
             return <BlockLoader />;
           }
 
-          const onSubmit = (e: FormEvent): void => {
+          const onSubmit = async (e: FormEvent): Promise<void> => {
```
Comment (EmiliaMakelaVincit) [inline src/areaSearch/areaSearchSpecsPage.tsx:181]: Hmm, is this necessary? Considering that this doesn't return anything, I'm not sure what's the effective difference between sync and async here.

File: src/areaSearch/requests.ts [hunk 7]
```diff
@@ -1,27 +1,68 @@
...
     formData.append(key, '' + rest[key]);
   });
+
   formData.append('geometry', JSON.stringify(geometry));
-  attachments.forEach((file, i) => formData.append(`attachments[${i}]`, file));
+  // formData.append('area_search_attachments', JSON.stringify(area_search_attachments));
 
+  
+  area_search_attachments.forEach((file, i) => formData.append(`area_search_attachments[${i}]`, `${file}`));
+  */
```
Comment (EmiliaMakelaVincit) [inline src/areaSearch/requests.ts:44]: Can be removed.

File: src/application/components/_applicationFractionalField.scss (hunk 4 above)
Comment (PasiVuohijoki) [reply src/application/components/_applicationFractionalField.scss:5]: I just checked with inspector that these were the properties causing ugly extra margin on the outer sides of the fields. :laughing: So I cancelled those by setting to 0 and it fixed the issue..

**Review by EmiliaMakelaVincit — COMMENTED** [review]
Besides the SCSS class stuff from the last review, this looks good to me now.

**Review by EmiliaMakelaVincit — APPROVED** [review]

**Pull request merged**